Project-Descriptions-and-Plans/CV32E4_TBD/Project_Concept_for_CV32E4_TBD_Nov-27_2023.md

```diff
@@ -0,0 +1,250 @@
+  external interrupt controller that requires a 2-step interrupt handling code.
+
+Design Portability:
+* Replace global clock gate by equivalent global enable signal
+  * this will make the design more "FPGA-friendly", or compliant with ASIC design rules that forbid any logic addition on clock tree
+  * clock gate cell model and/or simulator behaviour may introduce delta cycles on clock tree, leading to RTL/GL simulation mismatch
+* EDA tools parser bug work-arounds: current design coding style is standard-compliant, but EDA tools may
+  have bugs that prevent proper reading. Design should be parsed with several EDA tools and offending RTL code should be fixed.
+
+### Potential future design enhancements
+* remove deprecated features
+* Add Zce extension for further code-size reduction (not required for Multicore cluster).
+* Add bus attributes support.
+* move bitmanip, SIMD and complex number subsets of XPULP, RVB and RVP to a CV-X-IF compliant co-processor IP
+  * pros : this will make these features available to other processors, and would simplify verification work.
+  * cons : a performance loss is expected on instructions that use register pairs.
+
+## Who would make use of OpenHW output
+Companies needing more performances, less energy consumption or smaller code size for real-time control and/or DSP applications.
+
+## Initial Estimate of Timeline
+At Project Concept:
+* Preliminary RTL release expected in 2024 Q1
+* RTL feature-complete expected in end of 2024 Q4
+* RTL freeze and verification completed expected in 2025 Q4
+
+## Explanation of why OpenHW should do this project
+This proposal is a continuation of CV32E40P project.
+Main goal is to support official RVB (and a part of RVK) and (upcoming) RVP
+extensions while keeping backward compatibility with CV32E40Pv1/v2.
+One key advantage of supporting official RVB/RVP extensions is versatility and
+compatibility with a larger set of framework and applications. This will also allow
+users to use upstream version of compilation toolchain.
+
+XPULP custom extensions may also be extended to benefit from design enhancements
+introduced for RVB and RVP support.
+
+Additionnal features may also be considered, like CV-X-IF and CLIC support.
+
+Improved OBI interface support (bus error in particular) is also envisionned.
+
+Keeping backward compatibility will allow existing users of CV32E40Pv2 to migrate
+smoothly their existing products to this new version of the core, while keeping
+the benefits of existing custom extensions.
+In addition, keeping XPULP custom extensions will offer improvements to address
+manipulation, flow control and complex numbers computations.
+
+## Industry landscape: description of competing, alternative, or related efforts in the industry
+ARM Cortex M4
+
+## OpenHW Members/Participants committed to participate
+* CEA - RVB/RVP/XPLULP Design & Verification
```

Comment:
What can we do to bring other OpenHW members into participation? We don't normally want a project with just one member driving. That can be a PL deliverable item.

Reply:
Davide and his colleagues from EPFL intend to work on CV-X-IF. Work has started already here : https://github.com/esl-epfl/cv32e40px
I don't know yet if Dolphin or other partners would be interested to participate.